### PR TITLE
Form Control Enhancements - Part 2

### PR DESCRIPTION
**Pull request by jbouder**
In order to simplify the developer experience when using form components, this PR is to include a base wrapper (using the Form Group) with all applicable form components, allowing the developer provide label, helper text, errors, required attribute, and validation status directly through the form control component.

## Description

- Updated select, combobox, checkbox group, radio button group, date picker, file input, char count, memorable date, time picker, and range slider with form group wrapper
- Updated form group to handle children form fields with wrapper elements
- Updated contact form with more form components
- Updated stories and unit tests based on new wrapper use
- Misc fixes and updates
- Styling update to simplify displaying form components in error state

## Related Issue

- Includes fix for #241 

## Motivation and Context

- Simplified Developer Experience
- Encourage standardized form component layout

## How Has This Been Tested?

- Local Testing
- Beta Release Testing in Comet Starter

## Screenshots (if appropriate):
Screenshot 2024-08-21 at 8 14 54 AM

### CODE REVIEW

File: packages/comet-uswds/src/components/range-slider/range-slider.tsx [hunk 1]
```diff
@@ -42,17 +64,26 @@ export const RangeSlider = ({
...
+          min={minValue}
+          max={maxValue}
+          step={step}
+          defaultValue={defaultValue}
+          onChange={onChange}
+          role="slider"
+          {...inputProps}
+        />
+      }
     />
```
Comment (kildre) [inline packages/comet-uswds/src/components/range-slider/range-slider.tsx:87]: What if there were an option to display the minimum and max range on the slider, then maybe the output of where you are at on the range, a number that always floated above the slider circle?

File: packages/comet-uswds/src/components/text-area/text-area.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import classnames from 'classnames';
 import React, { ChangeEventHandler } from 'react';
-import { ValidationStatus } from 'src/utils/types';
+import { ValidationStatus } from '../utils/types';
 import FormGroup from '../form-group';
 
 export interface TextAreaProps {
```
Comment (kildre) [inline packages/comet-uswds/src/components/text-area/text-area.tsx:6]: Maybe add an optional character counter and an optional max character prop?

File: packages/comet-uswds/src/components/time-picker/time-picker.tsx [hunk 3]
```diff
@@ -48,22 +70,33 @@ export const TimePicker = ({
...
+        const inputElement = timePickerElement.querySelector(
+          '.usa-combo-box__input',
+        ) as HTMLInputElement;
+        if (onChange) {
+          inputElement.removeEventListener('change', onChange);
+        }
       }
-      comboBox.off(timePickerElement);
-      timePicker.off(timePickerElement);
     };
```
Comment (kildre) [inline packages/comet-uswds/src/components/time-picker/time-picker.tsx:100]: Maybe update the display format from 8:32am to 8:32 a.m.

Comment (jbouder) [reply packages/comet-uswds/src/components/time-picker/time-picker.tsx:100]: This is using the format provided by USWDS so I htink we should leave it as is

File: packages/comet-uswds/src/components/date-picker/date-picker.tsx [hunk 4]
```diff
@@ -43,10 +66,16 @@ interface DatePickerAttributes {
...
   maxDate,
   dateRange,
   defaultValue,
+  required,
+  label,
+  helperText,
+  errors,
+  validationStatus,
   onChange,
   ...inputProps
```
[the remark this answers is not in the document]
Comment (jbouder) [reply packages/comet-uswds/src/components/date-picker/date-picker.tsx:80]: This component includes all props supported by an HTML input, based on this line of code: ` DatePickerProps & JSX.IntrinsicElements['input']`, so those values could be provided as needed.

File: packages/comet-uswds/src/components/file-input/file-input.tsx [hunk 5]
```diff
@@ -10,21 +11,53 @@ export interface FileInputProps {
...
+  label?: string;
+  /**
+   * Helper text to display with the input
+   */
+  helperText?: string;
+  /**
+   * An array of string error messages
+   */
+  errors?: string | string[];
 }
```
[the remark this answers is not in the document]
Comment (jbouder) [reply packages/comet-uswds/src/components/file-input/file-input.tsx:30]: This component includes all props supported by an HTML input, based on this line of code: `FileInputProps & JSX.IntrinsicElements['input']`, so the maxSize prop could be passed in as needed.

Loading isn't something thats supported by the input though, I think that would be better handled by a combination of components (FileInput and Loading).

File: packages/comet-uswds/src/components/range-slider/range-slider.tsx (hunk 1 above)
Comment (jbouder) [reply packages/comet-uswds/src/components/range-slider/range-slider.tsx:87]: I think is is something that could be implemented on an as needed basis with a combination of components. If did want to add that, we could handle through another PR though.

File: packages/comet-uswds/src/components/text-area/text-area.tsx (hunk 2 above)
Comment (jbouder) [reply packages/comet-uswds/src/components/text-area/text-area.tsx:6]: This component includes all props supported by an HTML input, based on this line of code: `TextAreaProps & JSX.IntrinsicElements['textarea']`, so the maxLength prop could be added as needed.

And I think adding Character Count as a separate component as needed would be best.

File: packages/comet-uswds/src/components/memorable-date/memorable-date.tsx [hunk 6]
```diff
@@ -45,71 +66,80 @@ export const MemorableDate = ({
...
+              className="usa-input"
+              id={`memorable-date-day-${id}`}
+              name={`memorable-date-day-${id}`}
+              minLength={1}
+              maxLength={2}
+              pattern="[0-9]*"
+              inputMode="numeric"
+              defaultValue={defaultDayValue}
+              onChange={onDayChange}
+            />
```
[the remark this answers is not in the document]
Comment (jbouder) [reply packages/comet-uswds/src/components/memorable-date/memorable-date.tsx:123]: Fixed

File: packages/comet-uswds/src/components/memorable-date/memorable-date.tsx [hunk 7]
```diff
@@ -45,71 +66,80 @@ export const MemorableDate = ({
...
+              className="usa-input"
+              id={`memorable-date-year-${id}`}
+              name={`memorable-date-year-${id}`}
+              minLength={4}
+              maxLength={4}
+              pattern="[0-9]*"
+              inputMode="numeric"
+              defaultValue={defaultYearValue}
+              onChange={onYearChange}
+            />
```
[the remark this answers is not in the document]
Comment (jbouder) [reply packages/comet-uswds/src/components/memorable-date/memorable-date.tsx:140]: Fixed

**Review by kildre — APPROVED** [review]
Approved

**Pull request merged**